src/modes/ModeEstimation.jl

```diff
@@ -0,0 +1,301 @@
+using ..Turing
+using ..Bijectors
+using LinearAlgebra
+
+import ..AbstractMCMC: AbstractSampler
+import ..DynamicPPL
+import ..DynamicPPL: Model, AbstractContext, VarInfo, AbstractContext, VarName,
+    _getindex, getsym, getfield, settrans!,  setorder!,
+    get_and_set_val!, istrans, tilde, dot_tilde
+import Optim
+import Optim: optimize
+import NamedArrays
```

Comment:
Optional dependencies have to be loaded as 
```suggestion
import .Optim
import .Optim: optimize
import .NamedArrays
```
otherwise users will get a warning about them not being listed as dependencies of Turing.